### PR TITLE
Implement 2,000 returned results for partitioned search queries

### PR DESCRIPTION
**Pull request by jiangphcn**
## Overview
Implement 2,000 returned results for partitioned search queries

## Testing recommendations
https://github.com/cloudant-labs/dreyfus/pull/38

## Related Issues or Pull Requests
https://github.ibm.com/cloudant/partition-queries/issues/44

## Checklist

- [X] Code is written and works correctly;
- [X] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

File: src/dreyfus_httpd.erl [hunk 1]
```diff
@@ -443,6 +444,13 @@ validate_search_restrictions(Db, Args) ->
             ok
     end,
 
+    case fabric_util:is_partitioned(Db) of
```
Comment (garrensmith) [inline src/dreyfus_httpd.erl:453]: Maybe we should put these into the config so that we can make them customizable rather than hardcoded like this.

Comment (jiangphcn) [reply src/dreyfus_httpd.erl:453]: Originally, `max_limit` is hardcoded, and followed the same way for `max_limit_partitions`. However, good idea to make it configurable and changed with `8caff576768aeabdf38b7e48aaaf67078e849ff3`

**Review by garrensmith — APPROVED** [review]
Nice work

File: src/dreyfus_httpd.erl [hunk 2]
```diff
@@ -256,7 +262,7 @@ parse_index_param("bookmark", Value) ->
 parse_index_param("sort", Value) ->
     [{sort, ?JSON_DECODE(Value)}];
 parse_index_param("limit", Value) ->
-    [{limit, parse_non_negative_int_param("limit", Value, "max_limit", "200")}];
+    [{limit, ?JSON_DECODE(Value)}];
```
Comment (rnewson) [inline src/dreyfus_httpd.erl:265]: I don't think we should broaden this here.

Comment (jiangphcn) [reply src/dreyfus_httpd.erl:265]: Reading that `parse_index_param/2` is for GET method and `parse_json_index_param/2` is for POST method. `limit` can be specified both in GET and POST method. For partitioned database, the allowed max is 2000 while non-partitioned database is 200. So I try to extract the specified limit here and validate in https://github.com/cloudant-labs/dreyfus/pull/35/files/edb5a513cbb6054f28069ec72d0481a1c419992e#diff-1ffd08cb305df4b1411fbfd64633c17bR456 and https://github.com/cloudant-labs/dreyfus/pull/35/files/edb5a513cbb6054f28069ec72d0481a1c419992e#diff-1ffd08cb305df4b1411fbfd64633c17bR460

**Comment by davisp** [comment]
This work is now included in #34

**Pull request closed without merging**